docs/decisions/08-abac-claims.md

```diff
@@ -0,0 +1,98 @@
+# Authorization Claims
+
+Letterbook makes use of a attribute based authorization scheme. Attributes are primarily (but likely not exclusively) expressed as claims. This describes that scheme, and an initial set of claims to work from.
+
+## Status
+
+- [ ] Decided
+- [ ] Decided Against
+- [ ] Deferred
+- [ ] Superseded
+
+# Decision
+
+A summary of what was proposed and decided
+
+## Claims
+
+The initial set of claims to work from
+
+Profile 
+- `Rename` (actions related to the display name and handle)
+- `Edit` (actions related to other fields like description and pfp)
+- `Follow` (follow other profiles)
+- `Unfollow` (stop following other propfiles)
+- `ApproveFollower` (approve, reject, or remove followers)
+- `Block` (actions related to moderation tools like block, mute, etc)
+- `ReadNotif` (view the notifications sent to the profile)
+- `ClearNotif` (mark notifications as read)
+- `Export` (actions related to exporting bulk data about the profile)
+- `Sign` (signing client-provided data with a server-managed key)
+- `ManageKeys` (add or revoke signing keys)
+- `Grant` (grant claims to other agents)
+- `Guest` (other claims are delegated, and some highly sensitive actions should still be prohibited)
+- `Owner` (holder is the owner, useful to help limit the size of tokens)
+- Post
+  - `Draft`
+  - `Publish`
+  - `Delete`
+  - `Read`
+  - `ReadDraft`
+  - `React`
+  - `Share`
+  - `SortReply`
+  - `ApproveReply`
+  - `Collect`
+  - Coauthor stuff?
+  - Mentions?
+  - Private mentions?
+  - Media stuff?
```

Comment:
I'm not sure how these should be broken down, and maybe it's best to defer it to later? Coauthoring is a feature I really want to support, but it's going to be a while. Mention-related attributes seem like they would be helpful to moderators. You could, for instance, revoke someone's ability to send mastodon-style private messages, which I expect could curtail some spam and abuse.

But, maybe it should wait until we're actually building out moderation features?